ENT-330 Add enterprise landing page

zubair-arbi opened the pull request:
**WIP**
@saleem-latif @asadiqbal08 @mattdrayer @douglashall 

**Description:** Add static enterprise landing page with course mode selection.

**JIRA:** [ENT-330](https://openedx.atlassian.net/browse/ENT-330)

**Dependencies:** N/A

**Merge deadline:** 5 May, 2017

**Installation instructions:**  Install `edx-enterprise` from this branch `zub/ENT-330-enterprise-landing-page` in `edx-platform`.

**Testing instructions:**

1. Go to enterprise customers django admin at `admin/enterprise/enterprisecustomer/`
2. Add enterprise customer with logo
3. Add a course in the enterprise customer catalog
4. View the enterprise landing page for the enterprise course by accessing the url with enterprise customer uuid `ec_uuid` from step no 2, e.g: `http://zubair.localhost:8000/enterprise/course/course-v1:edX+TEST_01+2016_R1/?ec_uuid=72416e52-8c77-4860-9584-15e5b06220fb`

**Merge checklist:**

- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [x] CI build is green
- [x] Version bumped
- [x] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [x] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

zubair-arbi commented:
@saleem-latif @asadiqbal08 @douglashall Please review.
I have added testing instructions on this PR. I will add sandbox link as soon as I am able to create one.

saleem-latif commented on enterprise/views.py:
```diff
@@ -416,3 +419,121 @@ def post(self, request):
+        "opportunities from the world's best universities."
+    )
+    confirmation_text = _("Confirm your course")
+    starts_at_text = _("Starts")
```
I think it would be better to put all constant string in a dict, dictionary can be defined at the top of the file. Thoughts ?

douglashall replied:
Better yet, why not put this string in the template?

saleem-latif commented on enterprise/views.py:
```diff
@@ -416,3 +419,121 @@ def post(self, request):
+        except (ValueError, EnterpriseCustomer.DoesNotExist):
+            raise Http404
+
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
```
since, we are using these variables only once, why not skip variable definitions and pass context data directly.
e.g.
```
context_data = {
     'page_title': _('Choose Your Track'),
     'page_language': get_language_from_request(request),
     'platform_name': configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME),
     'course_id': course_id,
    'course_name': course_details['name'],
    'course_organization': course_details['org'],
     ...
```
I think this will simply it.

douglashall commented on enterprise/urls.py:
```diff
@@ -15,6 +17,10 @@
         name='grant_data_sharing_permissions'
     ),
     url(
+        r'^enterprise/course/{}/$'.format(settings.COURSE_ID_PATTERN),
```
I'm wondering if the URL pattern here should be something more like `r'^enterprise/(?P<enterprise_uuid>[^/]+)/course/{}/enroll/$'.format(settings.COURSE_ID_PATTERN)`? @brittneyexline @mattdrayer @zubair-arbi Thoughts?

zubair-arbi replied:
@douglashall I like your suggested url pattern. As for us both `enterprise_uuid` and `course_id` are required.

douglashall commented on enterprise/views.py:
```diff
@@ -416,3 +419,121 @@ def post(self, request):
         return self.post_account_consent(request, consent_provided)
+
+
+class LandingPageView(View):
```
Should this actually be called something like `CourseEnrollmentView`?

douglashall commented on enterprise/views.py:
```diff
@@ -416,3 +419,121 @@ def post(self, request):
+            # pylint: disable=no-member
+            enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+        except (ValueError, EnterpriseCustomer.DoesNotExist):
+            raise Http404
```
I think we want to render the template with appropriate error messaging for each of the conditions above instead of returning the `Http404`.

douglashall commented on enterprise/views.py:
```diff
@@ -416,3 +419,121 @@ def post(self, request):
+            ),
+            'not_eligible_for_certificate_text': self.not_eligible_for_cert_text,
+            'continue_link_text': self.continue_link_text,
+        }
```
Yes, I would definitely move all of the static text you are passing here in the context to the template unless there is a good reason not to.

zubair-arbi replied:
@douglashall I agree that for now most of the content is just static string but the current implementation gives us flexibility. Here are my thoughts on this:
* Its easy to test the text content from the response context.
* Future manipulation of text (like adding dynamic values in the strings) can be done directly in the view without doing changes to the html. HTML gets hard to read when we add too much logic into the template.
* Easy to use different templating language without any extra effort.
* Consistent with the existing approach for the enterprise views
I agree that for now most of the content is just static string but this will give us flexibility. Please let me know your thoughts.

douglashall commented on enterprise/templates/enterprise/enterprise_landing_page.html:
```diff
@@ -0,0 +1,82 @@
+<%static:css group='${self.attr.main_css}'/>
+  <%!
+    from django.utils.translation import ugettext as _
+    %>
```
Please fix the indentation inconsistencies above.

zubair-arbi commented:
@saleem-latif @douglashall I have addressed most of your feedback. Please give another review.
I has also setup sandbox with enterprise landing page: https://zubair-arbi.sandbox.edx.org/enterprise/c62208d9-ab13-4116-aeb3-0a797d17839e/course/course-v1:HKUST+EC001+2017_R1/enroll/

chris-mike commented:
Overall, it looks pretty good. My primary concern is continuity between this page and the sign in / create account pages being created by @haikuginger and @pomegranited . Because we use the same left-column design for each, it's important that the size and dimension of what in the left column stay the same.

A few visual nits:
1. Although the main content is centered on the page, the edX logo should be wider than the content.  (see screenshot)
![ent-330-1](https://cloud.githubusercontent.com/assets/16578372/25759270/b2311818-319f-11e7-9106-f70ecb605c7f.png)
2. When I tab over the "continue" button, the state should change to the "on hover" (it should go darker, as when selected).
3. Please put a little more padding here - closer to 30 px.
![ent-330-2](https://cloud.githubusercontent.com/assets/16578372/25759917/83f7748a-31a2-11e7-949b-0a36b5b09a89.png)

4. Increase font size of "vandelay has partnered with..." to 16px
5. Increase size of vandelay logo to max height of 60px

zubair-arbi commented:
@chris-mike thanks for the detailed design review and suggestions.
1. I have adjusted the `enterprise-container` to have a max width of `700px`
2. Added css for tab over the "continue" button
3. Added padding for left and right section to `30px` w.r.t center line
4. Updated font size of enterprise welcome message to 16px
5. Increase size of enterprise logo to height of 60px
6. Added top margin to the course thumbnail so that it is top aligned with the name of the course
7. Hide enterprise logo when we are in mobile-responsive
You can view the updated page here (staff@example.com/edx): https://zubair-arbi.sandbox.edx.org/enterprise/c62208d9-ab13-4116-aeb3-0a797d17839e/course/course-v1:HKUST+EC001+2017_R1/enroll/

haikuginger commented:
FWIW, in Safari, the logo/blurb always appear stacked; never side-by-side:

![screen shot 2017-05-08 at 8 25 50 am](https://cloud.githubusercontent.com/assets/7773758/25804088/0c6a8a42-33c8-11e7-90b1-daedf5e0ed41.png)

zubair-arbi commented:
@haikuginger  thanks for pointing out the alignment issue on Safari. I have fixed the issue and updated the sandbox. Please have another look.

haikuginger commented:
Looks much better on Safari; thanks for the fix!

zubair-arbi commented:
@chris-mike Please let me know if you are ok with the current design.

chris-mike commented:
One tiny change: please remove the colon after "confirm your course."  Everything else looks good. Thanks!

👍
